agents/PCS-Large-LP-Deposit-Withdrawal/src/agent.ts

```diff
@@ -0,0 +1,52 @@
+import { BigNumber } from "ethers";
+import { Finding, HandleTransaction, TransactionEvent, LogDescription, getEthersProvider } from "forta-agent";
+import { EVENTS_ABI, POOL_SUPPLY_THRESHOLD, THRESHOLD_PERCENTAGE, FACTORY } from "./constants";
+import { createPair } from "./utils";
+import { createFinding } from "./finding";
+import PoolFetcher from "./pool.fetcher";
+const STKBNBPOOLADDRESS = "0xaa2527ff1893e0d40d4a454623d362b79e8bb7f1";
+
+export const provideHandleTransaction =
+  (
+    createPair: any,
+    fetcher: PoolFetcher,
+    poolSupplyThreshold: BigNumber,
+    thresholdPercentage: BigNumber,
+    factory: string
+  ): HandleTransaction =>
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const block: number = txEvent.blockNumber;
+    const logs: LogDescription[] = txEvent.filterLog(EVENTS_ABI);
```

Comment:
I think `filterLog` should allow you to filter even the logs for just the stkBNB pool address, and not all the logs from all the pools. Should look into this and optimize it, otherwise, this bot will be very slow.

So, when you go after the line:
```
if (!logs) return findings;
```

there should only be events that you really need to process, not just any random event happening on any other pool out there. That would just be a waste of computation.

Reply:
Done